spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :feature do
+  let(:user) { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+  let!(:post) { Post.new(author: user, title: 'Hello', text: 'This is my first post') }
+
+  before { user.save }
+
+  describe 'index page' do
+    before { visit user_posts_path(user) }
+
+    it 'shows the user name' do
+      expect(page).to have_content(user.name)
+    end
+
+    it "shows the user's profile picture" do
+      expect(page).to have_selector("img[src='#{user.photo}']")
+    end
+
+    it 'shows the number of posts for each user' do
+      expect(page).to have_content("Number of Posts: #{user.posts_counter}")
+    end
+
+    context 'when a recent post exists' do
+      let(:first_recent_post) { user.recent_post[0] }
+
+      it "shows the recent post's title" do
+        expect(page).to have_content(first_recent_post.title) if first_recent_post
+      end
+
+      it "shows some of the recent post's body" do
+        if first_recent_post
+          expected_text = if first_recent_post.text.length > 200
+                            "#{first_recent_post.text[0,
+                                                      200]}..."
+                          else
+                            first_recent_post.text
+                          end
+          expect(page).to have_content(expected_text)
+        end
+      end
+
+      it 'shows the number of comments for the recent post' do
+        expect(page).to have_content("Comments: #{first_recent_post.comments_counter}") if first_recent_post
+      end
+
+      it 'shows the number of likes for the recent post' do
+        expect(page).to have_content("Likes: #{first_recent_post.likes_counter}") if first_recent_post
+      end
+    end
+
+    it 'shows a section for pagination' do
+      expect(page).to have_selector('a', text: 'Pagination')
+    end
+  end
```

Comment:
There are 2 tests missing here for the post index page:

- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=the%20post%27s%20body.-,I%20can%20see%20the%20first%20comments%20on%20a%20post.,-I%20can%20see), Please add Tests that asserts you can see the first comments on a post.
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=on%20the%20view.-,When%20I%20click%20on%20a%20post%2C%20it%20redirects%20me%20to%20that%20post%27s%20show%20page.,-Post%20show%20page), Please add Tests that redirects to the post's show page when you click on a post.